### PR TITLE
TDI-36713 : TMarketoInput can not handle large input to match data

### PR DESCRIPTION
**Pull request by pyzhou-talend**
https://jira.talendforge.org/browse/TDI-36713
414 will be returned when the URI of a GET request exceeds 8KiB.  To avoid it, check against the length of your query string to see if it exceeds this limit.
If does it will be divided into smaller requests.

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tMarketoGenArray/tMarketoGenArray_end.javajet [hunk 1]
```diff
@@ -34,16 +34,17 @@
...
-						}else{
-							leadKeyValues_<%=cid %>.add(tmpBatch_<%=cid %>.subList(<%=segmentSize%>*arrayCount_<%=cid%>,tmpBatch_<%=cid %>.size()).toArray(new String[tmpBatch_<%=cid %>.size()-<%=segmentSize%>*arrayCount_<%=cid%>]));
-							break;
-						}
-					}
+							int count_<%=cid%> = 0,start_<%=cid%> = 0,end_<%=cid%> = 0;
+		                    for (String string : tmpBatch_<%=cid %>) {
+			                    count_<%=cid%> += java.net.URLEncoder.encode(string).length();
+			                    end_<%=cid%>++;
+			                    if(count_<%=cid%>>20){
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tMarketoGenArray/tMarketoGenArray_end.javajet]: What does "20" means? and how did you get this number?
I think your change is can be by fix by change the segment default size.
Did you check the solution which showed on jira comment ?
> If it does change your request to a POST method, then input your query string as the request body with the additional parameter ‘_method=GET’. This forgoes the limitation on URIs.

File: main/plugins/org.talend.designer.components.localprovider/components/tMarketoGenArray/tMarketoGenArray_end.javajet [hunk 2]
```diff
@@ -34,16 +34,17 @@
...
-							leadKeyValues_<%=cid %>.add(tmpBatch_<%=cid %>.subList(<%=segmentSize%>*arrayCount_<%=cid%>,<%=segmentSize%>*(arrayCount_<%=cid%>+1)).toArray(new String[<%=segmentSize%>]));
-							arrayCount_<%=cid%>++;
-						}else{
-							leadKeyValues_<%=cid %>.add(tmpBatch_<%=cid %>.subList(<%=segmentSize%>*arrayCount_<%=cid%>,tmpBatch_<%=cid %>.size()).toArray(new String[tmpBatch_<%=cid %>.size()-<%=segmentSize%>*arrayCount_<%=cid%>]));
-							break;
-						}
-					}
+							int count_<%=cid%> = 0,start_<%=cid%> = 0,end_<%=cid%> = 0;
+		                    for (String string : tmpBatch_<%=cid %>) {
+			                    count_<%=cid%> += java.net.URLEncoder.encode(string).length();
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tMarketoGenArray/tMarketoGenArray_end.javajet]: deprecated(string) is a deprecated method. You should change to another method.

**Comment by pyzhou-talend** [comment]
I removed the whole previous commit, cause change the way to fix it.

File: main/plugins/org.talend.designer.components.libs/libs_src/talend-marketo-rest/src/main/java/org/talend/marketo/MarketoRestClient.java [hunk 3]
```diff
@@ -150,6 +150,25 @@ public RequestResult executePostRequest(Class<?> resultClass,
         }
 
     }
+    
+    public RequestResult executePostRequest(Class<?> resultClass,
+            String postContent) throws Exception {
+    	
+    	webClient.type(MediaType.APPLICATION_FORM_URLENCODED_TYPE);
+        Response response = webClient.post(postContent);
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.libs/libs_src/talend-marketo-rest/src/main/java/org/talend/marketo/MarketoRestClient.java]: I thinks most of the code in 2 "executePostRequest" a duplicate.
Better reduce the duplicate code.

**Pull request merged**